packages/ui/src/components/va-form/VaForm.stories.ts

```diff
@@ -1,12 +1,286 @@
+    <va-form
+      role
+      immediate
+      :hideErrorMessages="false"
+    >
+      <va-input :rules="[value => !!value || 'input is required']"/>
+    </va-form>
+  `,
+})
+
+HideErrorMessages.play = async ({ step }) => {
+  const [firstDefinedInput, secondDefineInput] = document.querySelectorAll('[role] > div > div:nth-child(2)')
+
+  await step('Does not display validation error', async () => {
+    expect(firstDefinedInput).toBeDefined()
+  })
+
+  await step('Displays validation error', async () => {
+    expect(secondDefineInput).toBeUndefined()
+  })
+}
+
+export const HideErrors = () => ({
+  components: { VaForm, VaInput },
+  template: `
+    [true]
+    <va-form
+      role
+      immediate
+      hideErrors
+    >
+      <va-input :rules="[value => !!value || 'input is required']"/>
+    </va-form>
+    [false]
+    <va-form
+      role
+      immediate
+      :hideErrors="false"
+    >
+      <va-input :rules="[value => !!value || 'input is required']"/>
+    </va-form>
+  `,
+})
+
+HideErrors.play = async ({ step }) => {
+  const [firstInput, secondInput] = document.querySelectorAll('[role] > div')
+
+  await step('Does not highlight invalid input', async () => {
+    expect(firstInput).not.toHaveClass('va-input-wrapper--error')
+  })
+
+  await step('Highlights invalid input', async () => {
+    expect(secondInput).toHaveClass('va-input-wrapper--error')
+  })
+}
+
+export const Focus = () => ({
+  components: { VaForm, VaInput, VaButton },
+  template: `
+    <va-form ref="form">
+      <va-input />
+    </va-form>
+    <va-button @click="$refs.form.focus()">
+      Focus first input
+    </va-button>
+  `,
+})
+
+Focus.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const firstInput = canvas.getByRole('textbox', { name: '' }) as HTMLElement
+  const button = canvas.getByRole('button', { name: 'Focus first input' }) as HTMLElement
+
+  await step('Focusses first input', async () => {
+    await userEvent.click(button)
+    expect(firstInput).toHaveFocus()
+  })
+}
+
+export const FocusInvalid = () => ({
+  components: { VaForm, VaInput, VaButton },
+  template: `
+    <va-form ref="form">
+      <va-input error />
+    </va-form>
+    <va-button @click="$refs.form.focusInvalidField()">
+      Focus invalid
+    </va-button>
+  `,
+})
+
+FocusInvalid.play = async ({ canvasElement, step }) => {
+  const canvas = within(canvasElement)
+  const input = canvas.getByRole('textbox', { name: '' }) as HTMLElement
+  const button = canvas.getByRole('button', { name: 'Focus invalid' }) as HTMLElement
+
+  await step('Focusses invalid input', async () => {
+    await userEvent.click(button)
+    expect(input).toHaveFocus()
+  })
+}
+
+export const ValidateAndResetValidation = () => ({
+  components: { VaForm, VaInput, VaButton },
+  data: () => ({ input: 'do not reset' }),
+  template: `
+    <va-form ref="form">
+      <va-input v-model="input"/>
+      <va-input :rules="[() => 'error']"/>
```

Comment:
`:rules="[Boolean]"` is fine here, we don't have to check for validation message.